app/controllers/api_controller/provision_requests.rb

```diff
@@ -18,5 +18,21 @@ def create_resource_provision_requests(type, _id, data)
       MiqProvisionVirtWorkflow.from_ws(version_str, @auth_user_obj, template_fields, vm_fields, requester, tags,
                                        additional_values, ems_custom_attrs, miq_custom_attrs)
     end
+
+    def deny_resource_provision_requests(type, id, data)
+      provreq = resource_search(id, type, collection_class(:provision_requests))
+      raise BadRequestError, "A reason is required for this action" if data['reason'].nil?
+      reason = data['reason']
+      provreq.deny(@auth_user_obj.userid, reason)
+      provreq
```

Comment:
probably do the begin/rescue on the deny call, returning and action signature (success/failure), look at vms.rb for examples, probably something like,

begin
  provreq.deny(...)
  action_result(true, ....)
rescue => err
  action_result(false,   err.to_s)
end